components/script/dom/mediaquerylist.rs

```diff
@@ -137,7 +140,14 @@ impl WeakMediaQueryListVec {
     pub fn evaluate_and_report_changes(&self) {
         for mql in self.cell.borrow().iter() {
             if let MediaQueryListMatchState::Changed(_) = mql.root().unwrap().evaluate_changes() {
-                mql.root().unwrap().upcast::<EventTarget>().fire_event(atom!("change"));
+                let ev = MediaQueryListEvent::new(&mql.root().unwrap().global(),
```

Comment:
I believe having a `let mql = mql.root().unwrap();` above this line would save you a lot of the unnecessary rooting and unwrapping every time you want to make a reference to the `MediaQueryList`.